### PR TITLE
Update Symbol finalzer and description getter methods

### PR DESCRIPTION
**Pull request by clover2123**

### CODE REVIEW

File: src/builtins/BuiltinSymbol.cpp [hunk 1]
```diff
@@ -33,14 +33,16 @@ static Value builtinSymbolConstructor(ExecutionState& state, Value thisValue, si
     if (newTarget.hasValue()) {
         ErrorObject::throwBuiltinError(state, ErrorCode::TypeError, "illegal constructor Symbol");
     }
-    Optional<String*> descString = nullptr;
+
     // If description is undefined, let descString be undefined.
-    if (!(argc == 0 || argv[0].isUndefined())) {
+    // NOTE) we represent `undefined` as nullptr here
```
Comment (zherczeg) [inline src/builtins/BuiltinSymbol.cpp:38]: Typo: (NOTE)

File: src/runtime/Symbol.h [hunk 2]
```diff
@@ -21,31 +21,58 @@
...
+};
+
 class Symbol : public PointerValue {
 public:
     explicit Symbol(Optional<String*> desc = nullptr)
         : m_typeTag(POINTER_VALUE_SYMBOL_TAG_IN_DATA)
         , m_description(desc)
+        , m_finalizerData(nullptr)
     {
+        // nullptr of desc represents `undefined`
```
Comment (zherczeg) [inline src/runtime/Symbol.h:50]: I would add this comment to the member

File: src/runtime/Symbol.h [hunk 3]
```diff
@@ -21,31 +21,58 @@
...
     }
 
-    Optional<String*> description() const
+    String* descriptionString() const;
+    Value descriptionValue() const;
+
+    SymbolFinalizerData* finalizerData() const
     {
-        return m_description;
+        ASSERT(!!m_finalizerData);
```
Comment (zherczeg) [inline src/runtime/Symbol.h:58]: I don't see a `hasFinalizerData`, is there a way to check this somehow?

File: src/runtime/Symbol.cpp [hunk 4]
```diff
@@ -47,13 +58,127 @@ Symbol* Symbol::fromGlobalSymbolRegistry(VMInstance* vm, String* stringKey)
...
 }
 
+Value Symbol::keyForSymbol(VMInstance* vm, Symbol* sym)
+{
+    ASSERT(!!sym);
+    auto& list = vm->globalSymbolRegistry();
+
+    for (size_t i = 0; i < list.size(); i++) {
+        // If SameValue(e.[[Symbol]], sym) is true, return e.[[Key]].
+        if (list[i].symbol == sym) {
```
Comment (zherczeg) [inline src/runtime/Symbol.cpp:68]: Maybe it is defined in the specification, but is it true that always maximum of one result is possible?

**Review by zherczeg — APPROVED** [review]
Just some minor things.

File: src/builtins/BuiltinSymbol.cpp (hunk 1 above)
Comment (clover2123) [reply src/builtins/BuiltinSymbol.cpp]: Thank you for the review. fixed it.

File: src/runtime/Symbol.h (hunk 2 above)
Comment (clover2123) [reply src/runtime/Symbol.h]: fix it too

File: src/runtime/Symbol.h (hunk 3 above)
Comment (clover2123) [reply src/runtime/Symbol.h:57]: This `removeFinalizer` will be called when it has a valid data by GC (it is guaranteed)

File: src/runtime/Symbol.cpp (hunk 4 above)
Comment (clover2123) [reply src/runtime/Symbol.cpp:68]: Yes, only one unique Symbol (with each Symbol description) will be registered according to the spec.

**Pull request merged**